demos/imagenet_classification_demo/main.cpp

```diff
@@ -0,0 +1,508 @@
+                          0,
+                          std::min(modelInputResolution, image.cols - fromWidth),
+                          modelInputResolution);
+    } else {
+        int fromHeight = image.rows/2 - modelInputResolution/2;
+        imgROI = cv::Rect(0,
+                          fromHeight,
+                          modelInputResolution,
+                          std::min(modelInputResolution, image.rows - fromHeight));
+    }
+    image(imgROI).copyTo(image);
+}
+
+int main(int argc, char *argv[]) {
+    try {
+        if (!ParseAndCheckCommandLine(argc, argv)) {
+            return 0;
+        }
+
+        // -----------------------------------------Read input images-----------------------------------------
+        std::vector<std::string> imageNames;
+        std::vector<cv::Mat> inputImages;
+        parseInputFilesArguments(imageNames);
+        unsigned inputImagesCount = imageNames.size();
+        if (inputImagesCount == 0) throw std::runtime_error("No images provided");
+        std::sort(imageNames.begin(), imageNames.end());
+        for (size_t i = 0; i < inputImagesCount; i++) {
+            std::string& name = imageNames[i];
+            const cv::Mat& tmpImage = cv::imread(name);
+            if (tmpImage.data == nullptr) {
+                std::cerr << "Could not read image " << name << '\n';
+                imageNames.erase(imageNames.begin() + i);
+                inputImagesCount--;
+                i--;
+            }
+            else {
+                inputImages.push_back(tmpImage);
+                imageNames[i] = name.substr(name.rfind('/') + 1);
+            }
+        }
+        // ---------------------------------------------------------------------------------------------------
+
+        // ----------------------------------------Read image classes-----------------------------------------
+        bool hasClasses = !FLAGS_classes.empty();
+        std::vector<std::string> classIndexes;
+        if (hasClasses) {
+            std::vector<std::pair<std::string, std::string>> classIndexesMap;
+            std::ifstream inputClassesFile(FLAGS_classes);
+            while (true) {
+                std::string imageName;
+                std::string classIndex;
+                inputClassesFile >> imageName >> classIndex;
+                if (inputClassesFile.eof()) break;
+                classIndexesMap.push_back({imageName.substr(imageName.rfind('/') + 1), classIndex});
+            }
+            for (size_t i = 0; i < inputImagesCount; i++) {
+                for (size_t j = 0; j < classIndexesMap.size(); j++) {
+                    if (imageNames[i] == classIndexesMap[j].first) {
+                        classIndexes.push_back(classIndexesMap[j].second);
+                        break;
+                    }
+                }
+                if (i+1 != classIndexes.size()) {
+                    throw std::runtime_error("No class specified for image " + imageNames[i]);
+                }
```

Comment:
making `classIndexesMap` to be of `std::map` type will allow you to write `classIndexesMap.find(imageNames[i])` instead of the loop and the tricky check that the item was found.